panther_manager/include/panther_manager/moving_average.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef PANTHER_MANAGER_MOVING_AVERAGE_HPP_
+#define PANTHER_MANAGER_MOVING_AVERAGE_HPP_
+
+namespace panther_manager
+{
+
+template <typename T>
+class MovingAverage
+{
+public:
+  MovingAverage(const int window_size = 5) : window_size_(window_size), sum_(T(0)) {}
+
+  void roll(const T value)
+  {
+    if (values.size() >= window_size_) {
+      sum_ -= values.front();
+      values.pop_front();
+    }
+    values.push_back(value);
+    sum_ += value;
+  }
+
+  T get_average() const
+  {
+    if (values.size() == 0) {
+      return T(0);
+    }
+    return sum_ / static_cast<T>(values.size());
+  }
+
+private:
+  int window_size_;
```

Comment:
const won't work in here. If class instance is defined using default constructor (eg. in header file) it will be impossible to overwrite window_size later in cpp file.

Reply:
Using `std::unique_ptr` and dynamically creating the MovingAverage object solves the problem. IMO this is a more acceptable way than using `window_size_` which should be constant as mutable